api/src/identity-access-management/domain/models/PasswordExpirationToken.js

```diff
@@ -1,14 +1,21 @@
+import jsonwebtoken from 'jsonwebtoken';
+
 import { config } from '../../../shared/config.js';
 import { tokenService } from '../../../shared/domain/services/token-service.js';
+import { PasswordResetTokenExpired } from '../errors.js';
 
 export class PasswordExpirationToken {
   constructor({ userId }) {
     this.userId = userId;
   }
 
   static decode(token) {
-    const decoded = tokenService.getDecodedToken(token, config.authentication.secret);
-    if (!decoded) return new PasswordExpirationToken({});
+    let decoded;
+    try {
+      decoded = tokenService.getDecodedToken(token, config.authentication.secret);
+    } catch (error) {
+      if (error instanceof jsonwebtoken.TokenExpiredError) throw new PasswordResetTokenExpired();
+    }
 
     return new PasswordExpirationToken({ userId: decoded.user_id });
```

Comment:
Suggestion de modification : 
* pour que les erreurs autres que `jsonwebtoken.TokenExpiredError` ne soient pas  attrapées sans être traitées
* pour éviter d’utiliser un `let` avec `let decoded` si ce n’est pas nécessaire

```suggestion
    try {
      const decoded = tokenService.getDecodedToken(token, config.authentication.secret);
      return new PasswordExpirationToken({ userId: decoded.user_id });
    } catch (error) {
      if (error instanceof jsonwebtoken.TokenExpiredError) {
        throw new PasswordResetTokenExpired();
      }
      
      throw error;
    }
```

Reply:
Ok pour` const decoded` !
Pour les autres erreurs que `PasswordResetTokenExpired` il me semble que nous avions décidé de ne pas modifier le fonctionnement actuel, tel qu'attendu dans les tests existants  `it('returns empty object for invalid token', function () {`